### DISCUSSION TITLE
estimateSize option causes infinite loop

### DISCUSSION THREAD

**Discussion by kirgans** [TanStack/virtual#587] [Q&A]
Hi,

I just updated from 3 beta to alpha and I got this error : 
`Unhandled Runtime Error
Error: Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.`

this error can be avoided when I comment the estimateSize option when call the hook but I have inaccurate element sizes..
Any help would be much appreciated,
Thanks

**Comment by cojoclaudiu** [comment]
Same issue here, commenting the estimated size will fix the infinite loop.
@spacedev75 did you find any solution or fix for that?

I'm using "@tanstack/react-virtual": "^3.0.0-alpha.0"

**Reply by piecyk** [reply]
Please install latest beta, "@tanstack/react-virtual": "3.0.0-beta.62",

**Reply by kirgans** [reply]
yep, wrap it in a useCallback like this : 

`estimateSize: useCallback(() => 200, []),`

But we really need a large update on the docs and examples, the lib seems DEAD right now..

**Reply by piecyk** [reply]
Yep we do, fyi docs and examples are based on beta

**Reply by kirgans** [reply]
I know you are doing a lot of work and thanks for that ! 

So right now, you are recommending using beta instead of alpha ?

**Reply by piecyk** [reply]
Yes

**Reply by cojoclaudiu** [reply]
thank you @piecyk, @spacedev75 now I don't have anymore the infinite loop.
